src/Module.h

```diff
@@ -8,20 +8,34 @@
 
 #include "IR.h"
 #include "Buffer.h"
+#include "ModulusRemainder.h"
 #include "Outputs.h"
 #include "Target.h"
 
 namespace Halide {
 namespace Internal {
 
+struct LoweredArgument : public Argument {
```

Comment:
Should probably add an explicit comment about how LoweredArgument differs from Argument, and when you'd use it instead -- yes, this may seem obvious from the name but documentation is good :-)